M2/Macaulay2/packages/PrimaryDecomposition.m2

```diff
@@ -148,39 +147,34 @@ primaryComponent(Ideal, Ideal) := Ideal => opts -> (I, P) -> (primarycomponent (
 --------------------------------------------------------------------
 
 -- keys: none so far
-AssociatedPrimesOptions = new SelfInitializingType of BasicList
-AssociatedPrimesOptions.synonym = "associated primes options"
+AssociatedPrimesContext = new SelfInitializingType of Context
+AssociatedPrimesContext.synonym = "associated primes context"
 
 -- keys: CodimensionLimit and Result
-AssociatedPrimesComputation = new Type of MutableHashTable
+AssociatedPrimesComputation = new Type of Computation
 AssociatedPrimesComputation.synonym = "associated primes computation"
 
+new AssociatedPrimesContext from Ideal  :=
+new AssociatedPrimesContext from Module := (C, M) -> AssociatedPrimesContext{}
+
 new AssociatedPrimesComputation from Ideal  := (C, I) -> new C from comodule I
-new AssociatedPrimesComputation from Module := (C, M) -> (
-    cacheKey := AssociatedPrimesOptions{};
-    try M.cache#cacheKey else M.cache#cacheKey = new AssociatedPrimesComputation from {
-	Result => new MutableList from {}, CodimensionLimit => -1 })
+new AssociatedPrimesComputation from Module := (C, M) -> new AssociatedPrimesComputation from {
+    Result => new MutableList from {},
+    CodimensionLimit => -1 }
 
 -- TODO: make this unnecessary
```

Comment:
Do you have a plan to make this unnecessary?  What would need to be done?

Reply:
I haven't thought too much about whether this should be handled universally with something like `storeComputation` or separately for each computation like here. The former path would make this unnecessary.